lib/RestService/src/RestService.h

```diff
@@ -302,7 +302,8 @@ class RestService : public IService
     bool                          m_isWaitingForResponse;     /**< Is RestService still waiiting for a response to a request? */
     uint32_t                      m_activeRestId;             /**< Saves the  restId of a request until the callback triggered by the corresponding response is finished. */
     PreProcessCallback            m_activePreProcessCallback; /**< Saves the callback sent by a request until it is called when the response arrives. */
-    Mutex                         m_mutex;                    /**< Mutex to protect against concurrent access. */
+    mutable MutexRecursive        m_mutex;                    /**< Mutex to protect against concurrent access. */
```

Comment:
Why recursive or in other words, which public method is called during process() ? And couldn't it be avoided?